app/controllers/twilio_webhooks_controller.rb

```diff
@@ -38,12 +38,17 @@ def create_incoming_text_message
       )
 
       if FileTypeAllowedValidator::VALID_MIME_TYPES.include? content_type
-        extension = MIME::Types[content_type].first.extensions.first
-        filename_with_extension = "#{filename}.#{extension}"
-        document.upload.attach(io: StringIO.new(Net::HTTP.get(URI(media_url))),
-                                          filename: filename_with_extension,
-                                          content_type: content_type,
-                                          identify: false)
+        response = Net::HTTP.get_response(URI(media_url)) # first we get a redirect from Twilio to S3
+        response = Net::HTTP.get_response(URI(response['location'])) # then we get a redirect from S3 to S3
+        response = Net::HTTP.get_response(URI(response['location'])) # finally we should get a 200 OK with the file
```

Comment:
I'd prefer to see a call to `Net::HTTP` that follows all redirects. If that's available somehow and easy to use, then great. When Shannon I wrote `Net::HTTP.get()`, I was expecting that it would follow all redirects (not that I tested it).